base/doc/fntguide.tex

```diff
@@ -1997,6 +1997,63 @@ \subsection{Font series defaults per document family}
 would use |sb| (semi-bold) when |\rmfamily\bfseries| is requested in
 document.
 
+\subsection{Handling of current and requested font series and shape}
+
+In the original NFSS implementation, the series was a single attribute
+stored in |\f@series| and so one always had to specify both weight and
+width together.  Hence, it was impossible to typeset a paragraph in a
+condensed font and inside have a few words in bold weight (but still
+condensed) without doing this manually by requesting
+|\fontseries{bc}\selectfont|.
+
+\NEWfeature{2020/02/02}
+The new implementation now works differently by looking both at the
+current value of |\f@series| and the requested new series and out of
+that combination selects a resulting series value.  Thus, if the current
+series is |c| and we ask for |b|, we now get |bc|.  This is done by
+consulting a simple lookup table where entries can be added or changed
+with |\DeclareFontSeriesChangeRule|:
+\begin{decl}
+  |\DeclareFontSeriesChangeRule| %
+  \arg{current series} \arg{requested series} \\
+  \leavevmode\hfill    \arg{result} \arg{alternative result}
+\end{decl}
+
+The \arg{current series} is the value currently stored in |\f@series|,
+\arg{requested series} is the new series requested, \arg{result} is the
+combined value if it exists for the given font family, and
+\arg{alternative result} is a fallback in case \arg{result} doesn't
+exist.  The example above now looks like this:
+\begin{verbatim}
+   \DeclareFontSeriesChangeRule{c}{b}{bc}{}
+\end{verbatim}
+which means: switch to the |bc| series if |c| is current and |b| is
+(additionally) requested, and do nothing if the current font doesn't
+have the combination.
```

Comment:
The second part of the statement is not correct or at least not understandable. What happens is that  "c" + "b" unconditionally switches to "bc" but if that doesn't exist in the font then the normal font substitution kicks in, i.e., shape is switched back to "n" (usually) if that doesn't help series switches back to "m" (usually).

source2e says:

Only a few entries have “alternative” values and perhaps most of them should get dropped. Or maybe not . . . needs some thought perhaps.

>The idea is that you don’t want the normal substitution to kick in because that would reset the shape first and it may be better to stay with b when a change to c is requested and bc doesn’t exist, than to go to first change the shape to n and then find that bc/n doesn’t exist either and thus ending up with m/n.